### PR TITLE
Rollup of various changes on my side

### PR DESCRIPTION
**Pull request by promovicz**
This series of patches contains both some cleanups and some functional enhancements:

 * Support for overriding the Java version
 * Support for an archive JAR file
 * "Cleaner" handling of JCKit

Would you be willing to accept these as-is or should I separate the functional changes from the rest?

### CODE REVIEW

**Comment by martinpaljak** [comment]
All in all they look like good cleanups, give me some time to go through them.

Regarding File vs Path - there was something specific that made me decide to stick to Path-s, IIRC related to weird chars and Windows.

File: src/main/java/pro/javacard/ant/JavaCard.java [hunk 1]
```diff
@@ -287,6 +289,10 @@ public void setExport(String msg) {
             output_exp = msg;
         }
 
+        public void setJar(String msg) {
```
Comment (martinpaljak) [inline src/main/java/pro/javacard/ant/JavaCard.java]: Missing in README?

**Comment by promovicz** [comment]
Regarding File vs Path: I'll be investigating this and try to find out what your problem was. All in all Path is the more modern API. Too bad it conflicts with ant so badly. File seemed to yield simpler code, but maybe that was a mistake.

**Comment by martinpaljak** [comment]
Maybe it was a brainfart. I Usually leave a comment at the troublesome point, but as I do not see one, maybe it did not exist or got resolved. Thank you for the cleanup!

**Comment by martinpaljak** [comment]
Right. Try to run in a folder with space in the name.

**Comment by promovicz** [comment]
Actually couldn't leave this alone and took a look. The cause is not the use of File but how the files get passed as arguments to ant. Working on solution.

**Comment by martinpaljak** [comment]
FYI: After giving it some thought, I removed `javaversion` from `<javacard>`, as this feels like a very specific tweak, which should not "pollute" the absolute minimum required possible configurable, which is the javacard kit.

**Comment by gitfineon** [comment]
Hey @martinpaljak , tried to port some build stuff to `ant-javacard` but one project requires `javac -source 1.6 -target 1.6`. I was very pleased finding an option for that in your [README.md](https://github.com/martinpaljak/ant-javacard/blame/master/README.md#L80) but then found this comment.

Is there another way to force version? Because I get `unsupported class file format of version 52.0.` and this is exactly what happens when I remove those javac options in the old build file. My goal is to port without touching sources. But I'll take a look the "class file format" ... if it is a minor change.

**Comment by martinpaljak** [comment]
@gitfineon could you please clarify your scope and open a new issue ?

**Pull request merged**